frame/staking/src/pallet/mod.rs

```diff
@@ -1221,7 +1233,11 @@ pub mod pallet {
 			let controller = ensure_signed(origin)?;
 			let ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
 			let stash = &ledger.stash;
-			<Payee<T>>::insert(stash, payee);
+			Payee::<T>::insert(stash, payee.clone());
```

Comment:
I used a similar lazy migration approach in the paged rewards PR with some differences. Posting the strategy here as food for thought but I think current approach is also fine.

Would it be better if we not add new payee in this call and instead only to the new storage? A new function could abstract getter for `PayoutDestination` from either `Payee` or `Payees`. A Payee then would only exist in either one of storage at all times.

The migrate function (update_payee) would also delete the key from old storage (Payee) and move to new storage (Payees). That way, Payee would get empty and easy to kill (also potentially safer to know when migration is complete)?

Reply:
Downside to this is every call to `get the payee` would involve one extra storage read in worst case.

Reply:
This would be awesome to do, but it is concerning that `PayoutDestination` does not exist in any tests or benches yet. This new getter function would need to be embedded in all those tests and benches, and then removed again when the migration is complete. I think for the task at hand this is engineering overkill.